core/iwasm/interpreter/wasm.h

```diff
@@ -71,10 +71,6 @@ extern "C" {
 #define LABEL_TYPE_IF 2
 #define LABEL_TYPE_FUNCTION 3
 
-typedef struct WASMModule WASMModule;
```

Comment:
These three lines are introduced by multi-module feature, if removed, cmake .. -DWAMR_BUILD_MULTI_MODULE=1 && make will report error.

I have a test, we can remove the three lines, and modify some fields definition from WASMModule * to struct WASMModule *, e.g:
```c
typedef struct WASMTableImport {
    char *module_name;
    char *field_name;
    uint8 elem_type;
    uint32 flags;
    uint32 init_size;
    /* specified if (flags & 1), else it is 0x10000 */
    uint32 max_size;
#if WASM_ENABLE_MULTI_MODULE != 0
    WASMModule *import_module;   =>  struct WASMModule *import_module;
    WASMTable *import_table_linked; => struct WASMTable *import_table_linked;
#endif
} WASMTableImport;
```
Could you test whether it works in MacOS too?

Reply:
How about:
```
+ typedef struct WASMTableImport WASMTableImport;
- typedef struct WASMTableImport {
+ struct WASMTableImport {
    char *module_name;
    char *field_name;
    uint8 elem_type;
    uint32 flags;
    uint32 init_size;
    /* specified if (flags & 1), else it is 0x10000 */
    uint32 max_size;
#if WASM_ENABLE_MULTI_MODULE != 0
    WASMModule *import_module;
    WASMTable *import_table_linked;
#endif
- } WASMTableImport;
+ }
```
And it will keep WASMTableImport reference aligned: no need for a 'struct' always. Thought? @wenyongh

Reply:
Yes, seems it is a better way if it resolves the compilation issue?